lib/features/offline_mode/manager/opened_email_cache_manager.dart

```diff
@@ -62,33 +63,34 @@ class OpenedEmailCacheManager {
   }
 
   Future<bool> isOpenedDetailEmailCached(
-      AccountId accountId,
-      UserName userName,
-      DetailedEmail detailedEmail
+    AccountId accountId,
+    UserName userName,
+    EmailId emailId
   ) async {
-    final emailContentPathExists = await _isFileExisted(detailedEmail);
-    final detailedEmailCacheExists = await _isDetailEmailExistedInCache(accountId, userName, detailedEmail);
+    final emailContentPathExists = await _isFileExisted(emailId);
+    final detailedEmailCacheExists = await getDetailEmailExistedInCache(accountId, userName, emailId);
 
-    return emailContentPathExists == true && detailedEmailCacheExists;
+    return emailContentPathExists == true && detailedEmailCacheExists != null;
   }
 
-  Future<bool?> _isFileExisted(DetailedEmail detailedEmail) async {
+  Future<bool?> _isFileExisted(EmailId emailId) async {
     final fileSaved = await _fileUtils.isFileExisted(
-      nameFile: detailedEmail.emailId.asString,
-      folderPath: detailedEmail.folderPath,
+      nameFile: emailId.asString,
+      folderPath: CachingConstants.emailContentFolderName,
```

Comment:
it is make sure file exist and not empty?

Reply:
Yes, sureeee 

`
 /// Checks whether the file system entity with this path exists.
  ///
  /// Returns a `Future<bool>` that completes with the result.
  ///
  /// Since [FileSystemEntity] is abstract, every [FileSystemEntity] object
  /// is actually an instance of one of the subclasses [File],
  /// [Directory], and [Link]. Calling [exists] on an instance of one
  /// of these subclasses checks whether the object exists in the file
  /// system object exists *and* is of the correct type (file, directory,
  /// or link). To check whether a path points to an object on the
  /// file system, regardless of the object's type, use the [type]
  /// static method.
  Future<bool> exists();
`